cluster/src/main/java/io/scalecube/cluster/fdetector/FailureDetector.java

```diff
@@ -99,16 +93,32 @@ public FailureDetector(Transport transport) {
    * @param transport transport
    * @param config failure detector settings
    */
-  public FailureDetector(Transport transport, FailureDetectorConfig config) {
+  public FailureDetector(ITransport transport, FailureDetectorConfig config) {
     checkArgument(transport != null);
     checkArgument(config != null);
     this.transport = transport;
     this.config = config;
+    String nameFormat = "sc-failuredetector-" + transport.address().toString();
     this.executor = Executors.newSingleThreadScheduledExecutor(
-        new ThreadFactoryBuilder().setNameFormat("sc-failuredetector-%s").setDaemon(true).build());
+        new ThreadFactoryBuilder().setNameFormat(nameFormat).setDaemon(true).build());
     this.scheduler = Schedulers.from(executor);
   }
 
+  /**
+   * <b>NOTE:</b> this method is for testing purpose only.
+   */
+  ITransport getTransport() {
+    return transport;
+  }
+
+  /**
+   * <b>NOTE:</b> this method is for testing purpose only.
+   */
+  void resetSubject() {
```

Comment:
Lets remove it and instead listen for FD events in test until it will check specific member and see what is the result of this check.